Source/Android/jni/MainAndroid.cpp

```diff
@@ -566,10 +568,17 @@ JNIEXPORT void JNICALL Java_org_dolphinemu_dolphinemu_NativeLibrary_Initialize(J
   AchievementManager::GetInstance().Init(nullptr);
 
   // Start a background thread to periodically update controller input (every 10ms).
-  // This ensures Android keeps controller state updated even without a dedicated input loop.
+  // The thread checks `s_background_execution_allowed` and pauses updates when
+  // background execution is disallowed (to save battery).
   std::thread([]() {
     while (true)
     {
+      if (!s_background_execution_allowed.load())
+      {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        continue;
+      }
```

Comment:
This will still waste battery power by not allowing the CPU to go to sleep. When background execution isn't allowed, you need to either stop the thread entirely, or do a wait that won't return until background execution is allowed again.